### ISSUE TITLE
[Bug&Request]I can't hear the ATS plug-in sound from the second car onwards.

### ISSUE THREAD

**Issue by ryutazx1370** [leezer3/OpenBVE#844]
In the Train data that I made, There is a symptom that the sound is not output from the second car onwards.
There is a variable called 「PlayCarSoundDelegate」 in the ATS plug-in source, and it is made using it.
It would be helpful if you could confirm.

This symptom occurred after v1.8.4.3.

Also,I hear a noise from the driver's seat, Is it possible to make a sound come out from the middle?
There is also the joy of...

https://user-images.githubusercontent.com/120869823/208278894-08e5d6a1-f00c-4ff7-aef8-1fd3dd388996.mp4

**Comment by leezer3** [comment]
It doesn't appear to be broken here at first look, which would initially suggest that something is wrong in your plugin, or that you're doing something I didn't expect :)
Any chance of a look at the whole thing?

To set sound positions, please replace the sound.cfg with sound.xml:
https://openbve-project.net/documentation_hugo/en/trains/xml/sound.html

Not sure what you're trying to show with the video, please give more details.

**Comment by ryutazx1370** [comment]
It doesn't appear to be broken here at first look, which would initially suggest that something is wrong in your plugin, or that you're doing something I didn't expect :)
Any chance of a look at the whole thing?

>As for the source file, please refer to Sounds1-1.cs～ Sounds1-12.cs and Engine.cs in the attached file.
[Souce.zip](https://github.com/leezer3/OpenBVE/files/10268271/Souce.zip)
「Engine.cs」 is a source that contains the necessary conditions to produce engine sounds.
From each condition, it is made so that the sound of the ATS plug-in of the 1st to 12th cars can be produced.
for example,「this.Train.Sounds1c2.EngineLoop.Play(vol[0], pit[0], true, 1);」
If so, it is made so that the engine sound of the second car can be produced.

Not sure what you're trying to show with the video, please give more details.
>With the latest version, v1.8.4.3, there is no sound from the ATS plug-in for the second vehicle.
 On the other hand, before 1.8.4.2, for some reason, the sound comes out properly.

That's it, please let me know if it's not what you intended.

**Comment by leezer3** [comment]
Yep, you're definitely doing something a little more unexpected, which only originally worked due to inconsistancies / bugs.
Going to have to think about this some to avoid doing this again.

As a workaround, for each sound, you'll unfortunately need a distinct per-car copy (same file is fine, just needs a different ATS index)
Trigger this and it should be OK I think.

The commit which breaks things is the following:
https://github.com/leezer3/OpenBVE/commit/cc35338ceb9312898ddd379af176264c6efb4018

**TLDR:**
When designing this originally, it was in response to someone trying to play an announcement in a specific car.
This led to the assumption that only a single distinct copy of any given sound can be playing at any time, which seems reasonable enough.

The commit above fixes an edge case where the pitch / volume of a looped CarSound (used by plugins) are updated without stopping it first, which accidentally caused a duplicated sound.
It's just showing on car 1 & 2 due to timing (the stop / start commands are issued across separate frames)

**Comment by leezer3** [comment]
FWIW, the 'best' solution would probably be to add an API overload method to the PlayCarSound function to allow it to duplicate between cars, but if I can help it I'd prefer to keep the plugin API stable (backward compatability is important) :/

**Comment by leezer3** [comment]
Having thought about this a little more and looked at implementation tweaks, I've made the decision that for consistancy, a single sound index **must** only refer to a single distinct copy of a sound, and an appropriate warning has been added to the documentation.

The 'glitch' you're seeing where the sound seemingly works on cars 1 + 2 is just a quirk of timing, and isn't going to be changed at this point. 
(The stop command takes a frame to take effect, by which time a start command has already been issued for the next)
